xapian-applications/omega/postprocess

```diff
@@ -0,0 +1,247 @@
+
+    821f03288846297c2cf43c34766a38f7,book
+    098f6bcd4621d373cade4e832627b4f6,test
+    """
+    with open(final_log, 'r') as s, open(query_file, 'w+') as w:
+        reader = csv.DictReader(s)
+        writer = csv.writer(w)
+
+        for row in reader:
+            writer.writerow([row['QueryID'], row['Query']])
+
+
+def test_functions():
+    with open('test_search.log', 'w+') as test_search:
+        test_search.write('821f03288846297c2cf43c34766a38f7,"book","45,54",0\n')
+        test_search.write('821f03288846297c2cf43c34766a38f7,"book","45,54",0\n')
+        test_search.write('d41d8cd98f00b204e9800998ecf8427e,"","",0\n')
+        test_search.write('098f6bcd4621d373cade4e832627b4f6,"test","35,47",0\n')
+
+    with open('test_clicks.log', 'w+') as test_clicks:
+        test_clicks.write('821f03288846297c2cf43c34766a38f7,54\n')
+        test_clicks.write('821f03288846297c2cf43c34766a38f7,54\n')
+        test_clicks.write('098f6bcd4621d373cade4e832627b4f6,35\n')
+
+    generate_combined_log('test_search.log', 'test_clicks.log', 'test_final.log')
+
+    # Test entries in final.log are correct.
+    with open('test_final.log', 'r') as final_f:
+        reader = csv.reader(final_f)
+
+        # Skip header row.
+        next(final_f)
+
+        for i, row in enumerate(reader):
+            if i == 0:
+                assert row == ['821f03288846297c2cf43c34766a38f7',
+                              'book', '45,54', '0', '45:0,54:2'], "Incorrect entry in final.log"
+            if i == 1:
+                assert row == ['098f6bcd4621d373cade4e832627b4f6',
+                              'test', '35,47', '0', '35:1,47:0'], "Incorrect entry in final.log"
+
+    generate_query_file('test_final.log', 'test_query.txt')
+
+    # Test entries in query.txt are correct.
+    with open('test_query.txt', 'r') as query_f:
+        reader = csv.reader(query_f)
+
+        for i, row in enumerate(query_f):
+            if i == 0:
+                assert row == '821f03288846297c2cf43c34766a38f7,book\n', "Incorrect entry in query.txt"
+            if i == 1:
+                assert row == '098f6bcd4621d373cade4e832627b4f6,test\n', "Incorrect entry in query.txt"
+
+    # Safely delete test log files.
+    try:
+        os.remove('test_search.log')
+    except OSError as e:
+        print("Error: %s - %s." % (e.filename,e.strerror))
+
+    try:
+        os.remove('test_clicks.log')
+    except OSError as e:
+        print("Error: %s - %s." % (e.filename,e.strerror))
+
+    try:
+        os.remove('test_final.log')
+    except OSError as e:
+        print("Error: %s - %s." % (e.filename,e.strerror))
+
+    try:
+        os.remove('test_query.txt')
+    except OSError as e:
+        print("Error: %s - %s." % (e.filename,e.strerror))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='''Postprocess click data files.
+
+        This script generates the final clickstream log file from input search and
```

Comment:
This string will get formatted in the `--help` output, so it's important you don't include unwanted characters in it. The leading spaces here causes the output to look very odd (there's a surplus newline at the end as well):

```
$ python3 ./xapian-applications/omega/postprocess --help
usage: postprocess [-h] [--test] search_log clicks_log final_log query_file

Postprocess click data files.

        This script generates the final clickstream log file from input search and
        click log files and creates Query file that can be used by Xapian Letor
        module for generating its training files.
        

positional arguments:
  search_log  Path to the search.log file.
  clicks_log  Path to the clicks.log file.
  final_log   Path to save final.log file.
  query_file  Path to save query.txt file.

optional arguments:
  -h, --help  show this help message and exit
  --test      Run tests for this script.
```